Lib/test/test_embed.py

```diff
@@ -343,19 +343,34 @@ def test_finalize_structseq(self):
         out, err = self.run_embedded_interpreter("test_repeated_init_exec", code)
         self.assertEqual(out, 'Tests passed\n' * INIT_LOOPS)
 
-    @support.skip_if_pgo_task
     def test_quickened_static_code_gets_unquickened_at_Py_FINALIZE(self):
         # https://github.com/python/cpython/issues/92031
-        code = """if 1:
-            from importlib._bootstrap import _handle_fromlist
-            import dis
-            for name in dis.opmap:
-                # quicken this frozen code object.
-                _handle_fromlist(dis, [name], lambda *args: None)
+        from dis import _all_opmap
+        resume = _all_opmap["RESUME"]
+        resume_quick = _all_opmap["RESUME_QUICK"]
+        code = f"""if 1:
+            import importlib._bootstrap
+            func = importlib._bootstrap._handle_fromlist
+            code = func.__code__
+
+            # Assert initially unquickened.
+            # Use sets to account for byte order.
+            if set(code._co_code_adaptive[:2]) != set([{resume}, 0]):
+                raise AssertionError()
+
+            # Warmup
+            for i in range(30):
```

Comment:
It might be nice to have a constant somewhere, rather than an hardcoded number.

I'm not sure what is the number of iterations required to specialize a bytecode. Python/specialize.c contains:

```
static inline int                                                                      
initial_counter_value(void) {                                                          
    /* Starting value for the counter.                                                 
     * This value needs to be not too low, otherwise                                   
     * it would cause excessive de-optimization.                                       
     * Neither should it be too high, or that would delay                              
     * de-optimization excessively when it is needed.                                  
     * A value around 50 seems to work, and we choose a                                
     * prime number to avoid artifacts.                                                
     */                                                                                
    return 53;                                                                         
}     
```

Include/internal/pycore_code.h contains ``#define ADAPTIVE_CACHE_BACKOFF 64``.

I don't know which number matters.

Reply:
The relevant number is 8:

https://github.com/python/cpython/blob/main/Include/internal/pycore_code.h#L95

64 is how frequently an individual adaptive bytecode attempts to specialize itself. 53 is how many misses it takes before a specialized instruction reverts back to the adaptive instruction. These are both only relevant after the quickening has occurred (after 8 calls to `_PyCode_Warmup`).